scheduler/src/cook/mesos/constraints.clj

```diff
@@ -186,11 +187,17 @@
 ;; Group host placement constraints
 
 (defn get-cotasks
-  [db group job]
+  [db group job cotask-cache]
   "Given a group and a job, finds the set of all the running instances that belong to the jobs in
    'group', but do not belong to 'job'"
-  (-> (group/group->running-task-set db group)
-      (clojure.set/difference (set (:job/instance job)))))
+  (let [group-uuid (:group/uuid group)
+        db-cotasks (cache/lookup (swap! cotask-cache
+                                        (fn [c]
```

Comment:
This function can be extracted and potentially reused (after the refactor to `running-task-ids-cache` suggested below).